File: oa-portal/problems/m46-broken-keyboard/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

/* works[c - 'a'] is true when that key still functions. */
int longestTypeableRun(const string& s, const vector<bool>& works) {
    // write your code here

    return 0;
}

int main() {
    static char buf[1000006];
    if (scanf("%s", buf) != 1) return 0;
    int k;
    if (scanf("%d", &k) != 1) return 0;
    vector<bool> works(26, false);
    for (int i = 0; i < k; i++) { char c[8]; scanf("%s", c); works[c[0] - 'a'] = true; }

    printf("%d\n", longestTypeableRun(buf, works));
    return 0;
}

File: oa-portal/problems/m46-broken-keyboard/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    /* works[c - 'a'] is true when that key still functions. */
    static int longestTypeableRun(String s, boolean[] works) {
        // write your code here

        return 0;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        String s = in.next();
        int k = in.nextInt();
        boolean[] works = new boolean[26];
        for (int i = 0; i < k; i++) works[in.next().charAt(0) - 'a'] = true;

        System.out.println(longestTypeableRun(s, works));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
